### ISSUE TITLE
Adding `uniq` to dependencies breaks null-ls in neovim

### ISSUE THREAD

**Issue by sondr3** [rrrene/credo#1052]
### Environment

* Credo version (`mix credo -v`): `1.7.0`
* Erlang/Elixir version (`elixir -v`): 
  ```sh
  Erlang/OTP 25 [erts-13.2.2.1] [source] [64-bit] [smp:10:10] [ds:10:10:10] [async-threads:1] [jit] [dtrace]

  Elixir 1.15.0 (compiled with Erlang/OTP 25)
  ```
* Operating system: `macOS Ventura 13.4.1 (22F82)`

### What were you trying to do?

Run `credo` in neovim with null-ls without errors. However, I have added [`uniq`](https://github.com/bitwalker/uniq) to my dependencies which causes it to always recompile it resulting in

```sh
❯ mix credo suggest --format json
==> uniq
Compiling 4 files (.ex)
Generated uniq app
{ ... }
```

which breaks null-ls and starts inserting the text before the JSON object into my buffers. For example, if I open `mix.exs` and save after `null-ls` starts this is the result:

```elixir
❯ cat mix.exs
==> uniq
Compiling 4 files (.ex)
Generated uniq app
defmodule CredoUniq.MixProject do
  [...]
end
```

### Expected outcome

Credo and `null-ls` runs without mangling my buffers.

### Actual outcome

Credo inserts compilation text into my buffers.

**Comment by bitwalker** [comment]
For future reference, this issue occurred due to use of compile-time configuration in the `mix.exs` file, so if someone observes this issue elsewhere, that's what to look out for. This specific instance with `uniq` is fixed in it's 0.6.0 release, so this issue can probably be closed.

**Comment by sondr3** [comment]
Yep, thank you for your quick help and fix, I’ll close this issue 😊
